### PR TITLE
Add attribute checker

### LINKED ISSUE #774: Enhance WITH statement attributes parse for each submitter
We should have a common function call to validate attributes for different submitters to use:

```
func ValidateAttributes(definition map[string]*AttrMeta, ir *IR) (map[string]*attribute, error)
```

`AttrMeta` is a struct defines the type and documents for the attribute:

```
struct AttrMeta {
    typ AttrType, // can be INT, STRING, LIST
    doc string,
}
```

The `definition` is a mapping to define supported attributes for current submitter. e.g. for tensorflow submitter:

```
{
    "train.epochs": &AttrMeta{AttrTypeInt, "number of epochs to train"},
    "model.optimizer": &AttrMeta{AttrTypeString, "the name of optimizer to use"}
}
```

the submitter's `codegen` should call `ValidateAttributes` to validate attributes In the `IR`. And the document of the model using current `codegen` will be auto-generated.

### PR DESCRIPTION
**Pull request by tonyyang-svail**
Fix #774 

Next step:
1. make all IRs use attribute checker.
1. Implement `GenerateTableInMarkdown` for document generation.

### CODE REVIEW

File: pkg/sql/codegen/attribute/attribute.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	// Float indicates the corresponding attribute is a float
+	Float
+	// String indicates the corresponding attribute is a string
+	String
+	// IntList indicates the corresponding attribute is a list of integers
+	IntList
+)
+
+// Dictionary contains the description of all attributes
+type Dictionary map[string]*Description
```
Comment (typhoonzero) [inline pkg/sql/codegen/attribute/attribute.go:38]: Maybe use a more meaningful name than `Dictionary`?

**Review by typhoonzero — APPROVED** [review]
LGTM

File: pkg/sql/codegen/attribute/attribute.go (hunk 1 above)
Comment (tonyyang-svail) [reply pkg/sql/codegen/attribute/attribute.go:38]: I was struggling with the naming as well. Any suggestions? :)

Comment (typhoonzero) [reply pkg/sql/codegen/attribute/attribute.go:38]: Like `HyperParamsDefDict` ?

**Pull request merged**